data-hibernate-jpa/src/main/java/io/micronaut/data/hibernate/operations/HibernateJpaOperations.java

```diff
@@ -177,14 +180,14 @@ protected void setParameter(CommonQueryContract query, int parameterIndex, Objec
 
     @Override
     protected void setParameterList(CommonQueryContract query, int parameterIndex, Collection<Object> value) {
-        query.setParameter(parameterIndex, value);
+        if (value == null) {
+            value = Collections.emptyList();
+        }
+        query.setParameterList(parameterIndex, value);
     }
 
     @Override
     protected void setParameterList(CommonQueryContract query, int parameterIndex, Collection<Object> value, Argument<?> argument) {
-        if (value == null) {
-            value = Collections.emptyList();
-        }
```

Comment:
`value` is unused, also I am not quite sure what this function is doing with `query.setParameter(parameterIndex, parameterIndex);`

Reply:
That's a bug. Please change that line to `query.setParameterList(parameterIndex, value);`.